tests/PHPStan/Analyser/data/call-user-func-array-php8.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace CallUserFuncArray;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @template T
+ * @param T $t
+ * @return T
+ */
+function generic($t) {
+	return $t;
+}
+
+function fun(): int
+{
+	return 3;
+}
+
+function fun3($i, $x, $y): int
+{
+	return 3;
+}
+
+class c {
+	static function m(): string
+	{
+		return 'hello';
+	}
+}
+
+class Foo {
+	function proxy() {
+		$params = [
+			'CallUserFuncArray\generic',
+			[123]
+		];
+
+		assertType('mixed', call_user_func_array(...$params));
+
+		$params = [
+			'CallUserFuncArray\generic',
+			123,
+			456
+		];
+
+		// cufa expect max 2 args
+		assertType('*NEVER*', call_user_func_array(...$params));
+		assertType('mixed', call_user_func(...$params));
+	}
+
+	/**
+	 * @param string $params,...
+	 */
+	function doVariadics(...$params) {
+		assertType('string', call_user_func_array('CallUserFuncArray\generic', $params));
+		assertType('string', call_user_func('CallUserFuncArray\generic', ...$params));
+		// because of named arguments support in php8 we have a different return type as in php7
+		// see https://phpstan.org/r/58c30346-9568-47ca-82e5-53b2fffda7d0
+		assertType('array<int|string, string>', call_user_func('CallUserFuncArray\generic', $params));
```

Comment:
this is the only part where the php8 test-case differs from the php7 case

Reply:
So instead of two huge files that are mostly the same, could you please change this a bit so that:

1) There's a single file with assertType with the same results that's verified on both PHP 7 and PHP 8
2) There are two small files with different asserts that are if-elsed based on PHP version.

So 3 files in total. This is going to be easier to maintain. Thank you.

Reply:
thx for the feedback. just separated the tests